internal/gamepaddb/gen.go

```diff
@@ -0,0 +1,190 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build ignore
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+)
+
+const license = `// Copyright 2024 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+`
+
+// gamecontrollerdb.txt is downloaded at https://github.com/mdqinc/SDL_GameControllerDB.
+
+// To update the database file, run:
+//
+//     curl --location --remote-name https://raw.githubusercontent.com/mdqinc/SDL_GameControllerDB/master/gamecontrollerdb.txt
+
+//go:embed gamecontrollerdb.txt
+var gameControllerDB []byte
+
+const dbTemplate string = `{{.License}}
+
+{{.DoNotEdit}}
+
+{{.BuildConstraints}}
+
+package gamepaddb
+
+import (
+	_ "embed"
+)
+
+//go:embed gamecontrollerdb_{{.FileNameSuffix}}.txt
+var controllerBytes []byte
+//go:embed gamecontrollerdb_glfw.txt
+var glfwControllerBytes []byte
+
+func init() {
+	if err := Update(controllerBytes); err != nil {
+		panic(err)
+	}
+
+    if err := Update(glfwControllerBytes); err != nil {
+        panic(err)
+	}
+}`
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	// Follow the standard comment rule (https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source).
+	doNotEdit := "// Code generated by gen.go using 'go generate'. DO NOT EDIT."
+
+	type gamePadPlatform struct {
+		filenameSuffix   string
+		buildConstraints string
+	}
+
+	platforms := map[string]gamePadPlatform{
+		"Windows": {
+			filenameSuffix:   "windows",
+			buildConstraints: "//go:build windows && !microsoftgdk",
+		},
+		"Mac OS X": {
+			filenameSuffix:   "macos_darwin",
+			buildConstraints: "//go:build darwin && !ios",
+		},
+		"Linux": {
+			filenameSuffix:   "linbsd",
+			buildConstraints: "//go:build (freebsd || (linux && !android) || netbsd || openbsd) && !nintendosdk && !playstation5",
+		},
+		"iOS": {
+			filenameSuffix:   "ios",
+			buildConstraints: "//go:build ios",
```

Comment:
We don't need this. The filename suffix is already the constraint.

Reply:
Are you suggesting I remove `buildConstraints` from the iOS and Android defintions here, and then modify the template to set the build directive to e.g. `//go:build {{.FileNameSuffix}}` in the case it's not set?

Reply:
I mean

```go
"iOS": {
			filenameSuffix:   "ios",
}
```

is fine. We don't have to modify the template. `db_ios.go` doesn't need the constaraints since the file name already indicates the constraint.

> If a file's name, after stripping the extension and a possible _test suffix, matches any of the following patterns:
>
> *_GOOS
> *_GOARCH
> *_GOOS_GOARCH

https://pkg.go.dev/cmd/go#hdr-Build_constraints